synapse/storage/databases/main/registration.py

```diff
@@ -1210,6 +1210,7 @@ def set_user_deactivated_status_txn(self, txn, user_id: str, deactivated: bool):
         self._invalidate_cache_and_stream(
             txn, self.get_user_deactivated_status, (user_id,)
         )
+        self._invalidate_cache_and_stream(txn, self.get_user_by_id, (user_id,))
         txn.call_after(self.is_guest.invalidate, (user_id,))
```

Comment:
Good catch! 👍 

I don't see why we're invalidating the guest status here, but it seems like it has always been that way.

Reply:
I had started with implementing unit tests. The tests had always failed. That was the reason for catch this bug.